<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Documentation for CFTbaRcodes &mdash; CFTbaRcodes 1.0.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
    <link rel="shortcut icon" href="_static/favicon.ico"/>
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="#" class="icon icon-home"> CFTbaRcodes
            <img src="_static/logo.svg" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                1.0.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Documentation for CFTbaRcodes</a></li>
</ul>
</div>
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">CFTbaRcodes</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home"></a> &raquo;</li>
      <li>Documentation for CFTbaRcodes</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <figure class="align-default">
<a class="reference external image-reference" href="index.html"><img alt="logo" height="150px" src="_images/logo.svg" width="300px" /></a>
</figure>
<p>A web application for understanding CFTR haplotypes. The software is <a class="reference external" href="https://cftbarcode.research.sickkids.ca/">available here</a></p>
<section id="documentation-for-cftbarcodes">
<h1>Documentation for CFTbaRcodes<a class="headerlink" href="#documentation-for-cftbarcodes" title="Permalink to this headline"></a></h1>
<p>The first panel of the software shows a plot of the <em>CFTR</em> gene.</p>
<figure class="align-default">
<img alt="_images/cftr_plot.png" src="_images/cftr_plot.png" />
</figure>
<p>Along the top of the figure, known variants are plotted as small vertical black lines. Different parts of the gene can be selected for further exploration. By default, all exons are selected. Based on the selection, a table of variants will be shown in a panel below.</p>
<figure class="align-default">
<img alt="_images/exon_variants.png" src="_images/exon_variants.png" />
</figure>
<p>This table lists information regarding the rsID, position, REF/ALT and frequency of the variant in the dataset. In addition to the table, the variants from the table plotted by coordinate. SNPs are drawn as blue circles and indels appear as orange diamonds. The allele frequency of each variant is indicated by the darkness of the colour (darker = more common).</p>
<figure class="align-default">
<img alt="_images/variant_plot.png" src="_images/variant_plot.png" />
</figure>
<p>Variants can be selected by either the checkbox on the table view or by clicking the shape in the plotted view. Selected variants appear in the below panel.</p>
<figure class="align-default">
<img alt="_images/selected_variants.png" src="_images/selected_variants.png" />
</figure>
<p>Variant information is displayed in the table. Variants can be tagged with one of a set of predefined symbols (shown in last column). After clicking on the “Redraw Plot” button, variants tagged with a symbol will be shown in a haplotype table:</p>
<figure class="align-default">
<img alt="_images/redraw_button.png" src="_images/redraw_button.png" />
</figure>
<figure class="align-default">
<img alt="_images/haplotype_table.png" src="_images/haplotype_table.png" />
</figure>
<p>The haplotype table shows a count and frequency for each unique haplotype observed in the dataset. Rare haplotypes are hidden to protect privacy. The “Redraw Plot” button also generates a <a class="reference external" href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0010207">textile plot</a>:</p>
<figure class="align-default">
<img alt="_images/textile_plot.png" src="_images/textile_plot.png" />
</figure>
<p>In this plot, variants are shown in columns, lines connect alleles that are found on the same haplotype. The thicker the line, the more frequent a pair of alleles is found on the same haplotype. The plot optimized the vertical position of each allele such that alleles more often found on the same haplotype are aligned horizontally. More details can be found in the <a class="reference external" href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0010207">textile plot paper</a>.</p>
<p>Selecting a haplotype in the haplotype table highlights the corresponding path on the plot in red. Hovering over each variant provides frequency information.</p>
<p>The code is <a class="reference external" href="https://github.com/strug-hub/CFTbaRcodes">freely available on GitHub</a></p>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Scott Mastromatteo.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>